src/mqtt_client.rs

```diff
@@ -108,6 +109,7 @@ where
             ReceivedPacket::PubAck(ack) => {
                 // No matter the status code the message is considered acknowledged at this point
                 self.session_state.handle_puback(ack.packet_identifier)?;
+                self.send_quota.saturating_add(1);
```

Comment:
As per [4.9](https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901251), don't you also need to replenish on PubComp and PubRec-Error?

Reply:
Indeed - I got distracted half way through implementation - thanks for pointing this out. Should be cleaned up now.